src/code/sys_math.c

```diff
@@ -1,15 +1,93 @@
+/**
+ * Maths library: two factorials, integer power, wrappers for libultra's sins and coss (the main ones used), and some
+ * random functions moved from OoT's z_actor
+ */
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/sys_math/func_80179300.s")
+static f32 sFactorialTbl[] = { 1.0f,    1.0f,     2.0f,      6.0f,       24.0f,       120.0f,      720.0f,
+                               5040.0f, 40320.0f, 362880.0f, 3628800.0f, 39916800.0f, 479001600.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/sys_math/func_80179400.s")
+// Rename to Math_FactorialF
+/**
+ * Takes a float, returns the factorial of it(s trunctation), iteratively
```

Comment:
```suggestion
 * Takes a float, returns the factorial of its truncation, iteratively
```

Reply:
I put the brackets because it's not obvious that it's the truncation rather than the float itself (apart from, I suppose, trivial considerations of domains if we call this `Factorial`).

Reply:
Oh so is it suppose to mean something like:
`Takes a float, returns the factorial of it (s32-trunctation when i = n), iteratively
Or am I mis-understanding again?

Reply:
Yeah, the loop implicitly takes the floor.

Reply:
Cool, just the way the brackets were made it look like it was a victim of a misfired find-replace script or something lol (cause awk would never do that to us)